react-components/src/architecture/concrete/observations/SaveObservationsCommand.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type IconType, toast } from '@cognite/cogs.js';
+import { type ButtonType } from '../../../components/Architecture/types';
+import { type TranslateKey } from '../../base/utilities/TranslateKey';
+import { Changes } from '../../base/domainObjectsHelpers/Changes';
+import { ObservationsCommand } from './ObservationsCommand';
+
+export class SaveObservationsCommand extends ObservationsCommand {
+  public override get icon(): IconType {
+    return 'Save';
+  }
+
+  public override get tooltip(): TranslateKey {
+    return { fallback: 'Publish observation changes' };
+  }
+
+  public override get buttonType(): ButtonType {
+    return 'primary';
+  }
+
+  public override get isEnabled(): boolean {
+    const observation = this.getObservationsDomainObject();
+
+    return (
+      observation !== undefined &&
+      (observation.hasPendingObservations() || observation.hasPendingDeletionObservations())
+    );
+  }
+
+  protected override invokeCore(): boolean {
+    const tool = this.getTool();
+    if (tool === undefined) {
+      return false;
+    }
+
+    const domainObject = this.getObservationsDomainObject();
+
+    void domainObject
+      ?.save()
+      .then(() => {
+        toast.success({ fallback: 'Successfully published changes' }.fallback);
+
+        const observation = this.getObservationsDomainObject();
+        observation?.notify(Changes.geometry);
```

Comment:
Maybe I said you should use the geometry. But if you like the code to be better, the geometry has not change here. You can issue another change, maybe a new one(add in Changes) for this and listen to this in the updated. I will guess it is the colors that should update?

Reply:
It both saves newly created observations and deletes the ones marked for removal. So the visualization points should be recreated. I think `geometry` is appropriate here